angular6/src/app/app.component.html

```diff
@@ -0,0 +1,20 @@
+<h1>Form Builder, Form Group, Form Control, Validators and ngsubmit</h1>
+
+<form class="form" [formGroup]="userForm" (ngSubmit)="outputForm()">
+  <div class="field" [ngClass]="{'invalid': userForm.get('username')?.invalid && (userForm.get('username')?.touched || userForm.get('username')?.dirty)}">
```

Comment:
Create a function that checks all these conditions and returns only a boolean to simplify reading the html code